riscv/src/continuations/bootloader.rs

```diff
@@ -25,8 +25,15 @@ pub const BOOTLOADER_SPECIFIC_INSTRUCTION_NAMES: [&str; 2] =
 pub fn bootloader_preamble() -> String {
     let mut preamble = r#"
     // ============== extra rules on memory when compiled with continuations =======================
-    // The first operation can't be a read (because we expect this page to be paged-in)
-    m_change * m_is_read' = 0;
+
+    col witness m_is_bootloader_write;
+    m_is_bootloader_write * (1 - m_is_bootloader_write) = 0;
+    m_is_read * m_is_bootloader_write = 0;
+    m_is_bootloader_write * m_is_write = 0;
+
+    // The first operation has to be a write by the bootloader
+    // TODO: Comment, and is this sound? Could we insert rows of nothing before the first read?
+    m_change * (m_is_write' + m_is_read') = 0;
```

Comment:
Not sound. After `m_change`, the prover could set all `m_is_*` flags to 0 and then do a normal write or read.